File: examples/python/transaction_status.py
"""Example: Transaction Status

Queries the status of a previously initiated M-Pesa transaction by
TransactionID or OriginalConversationID. Result arrives via ResultURL.

Requires: MPESA_CONSUMER_KEY, MPESA_CONSUMER_SECRET,
          MPESA_INITIATOR_NAME, MPESA_INITIATOR_PASSWORD
Verified against the sandbox.
"""

import os

from daraja import Mpesa

mpesa = Mpesa({
    "consumer_key": os.environ["MPESA_CONSUMER_KEY"],
    "consumer_secret": os.environ["MPESA_CONSUMER_SECRET"],
    "environment": "sandbox",
    "initiator_name": os.environ["MPESA_INITIATOR_NAME"],
    "initiator_password": os.environ["MPESA_INITIATOR_PASSWORD"],
})

response = mpesa.transaction_status({
    "CommandID": "TransactionStatusQuery",
    "TransactionID": "NLA00TEST",
    "PartyA": 174379,
    "IdentifierType": 4,  # 4 = shortcode
    "ResultURL": "https://example.com/status/result",
    "QueueTimeOutURL": "https://example.com/status/queue",
    "Remarks": "Status check",
})

print(f"ResponseCode: {response.ResponseCode}")
print(f"ResponseDescription: {response.ResponseDescription}")

File: examples/typescript/transaction_status.ts
// Example: Transaction Status
//
// Queries the status of a previously initiated M-Pesa transaction by
// TransactionID or OriginalConversationID. Result arrives via ResultURL.
//
// Requires: MPESA_CONSUMER_KEY, MPESA_CONSUMER_SECRET,
//           MPESA_INITIATOR_NAME, MPESA_INITIATOR_PASSWORD
// Verified against the sandbox.

import { Mpesa } from "@daraja-sdk/ts";

const mpesa = new Mpesa({
  consumerKey: process.env.MPESA_CONSUMER_KEY!,
  consumerSecret: process.env.MPESA_CONSUMER_SECRET!,
  environment: "sandbox",
  initiatorName: process.env.MPESA_INITIATOR_NAME!,
  initiatorPassword: process.env.MPESA_INITIATOR_PASSWORD!,
});

const response = await mpesa.transactionStatus.query({
  CommandID: "TransactionStatusQuery",
  TransactionID: "NLA00TEST",
  PartyA: 174379,
  IdentifierType: 4, // 4 = shortcode
  ResultURL: "https://example.com/status/result",
  QueueTimeOutURL: "https://example.com/status/queue",
  Remarks: "Status check",
});

console.log(`ResponseCode: ${response.ResponseCode}`);
console.log(`ResponseDescription: ${response.ResponseDescription}`);